File: web/src/app/_model/common-search-view.ts
export class CommonSearchView<T>{
    public pageIndex: number = 0;
    public totalItem: number = 0;
    public itemPerPage: number = 0;
    public total_plan_qty: number = 0;
    public total_actual_qty: number = 0;
    public total_diff_qty: number = 0;
    public datas: Array<T> = [];
} 

File: WebApi/api/api/ModelViews/CommonSearchView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace api.ModelViews
{
    public class CommonSearchView<T>
    {
        public int pageIndex { get; set; }
        public int totalItem { get; set; }
        public int itemPerPage { get; set; }
        public List<T> datas { get; set; }
    }
}